release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py

```diff
@@ -162,20 +143,18 @@ def to_tensor_iterator():
     if config["num_images_per_epoch"] % config["batch_size"]:
         # Assuming batches will respect epoch boundaries.
         num_steps_per_epoch += 1
-    for epoch in range(config["num_epochs"]):
-        epoch_start_time_s = time.perf_counter()
 
+    for epoch in range(config["num_epochs"]):
         tf_dataset = None
         if config["data_loader"] == TF_DATA:
             assert _tf_dataset is not None
             tf_dataset = _tf_dataset
         elif config["data_loader"] == RAY_DATA:
             assert dataset_shard is not None
-            tf_dataset = ray_dataset_to_tf_dataset(
-                dataset=dataset_shard,
+            tf_dataset = dataset_shard.to_tf(
```

Comment:
1. for comparison with torch, I think we also need to do `iter_torch_batches` when data_loader is RAY_DATA. maybe we should refactor the script structure. 
2. Why not use `iter_tf_batches` instead of `to_tf`? iter should be more efficient?

Reply:
1. Yes that's marked as TODO in this code.
2. I think to_tf is the recommended API.

Reply:
Could you help me understand why `to_tf` is more preferable? 
It looks to me that for the torch examples, we recommend using `iter_torch_batches`. What is the difference between TF and Torch?

Reply:
The [docs](https://docs.ray.io/en/master/data/api/data_iterator.html) recommend using `iter_torch_batches` for torch and `to_tf` for TF.

I think this was mainly just to reduce friction for users and better match the third-party interfaces for torch DataLoader and tf.data.Dataset. There's some more context [here](https://github.com/ray-project/ray/issues/29031).